sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_message_backcompat.py

```diff
@@ -97,25 +124,30 @@ def _can_settle_message(self):
         return True
 
     @property
-    def settled(self):
+    def settled(self) -> bool:
         if self.state == MessageState.ReceivedUnsettled:
             return False
         return True
 
-    def get_message_encoded_size(self):
+    def get_message_encoded_size(self) -> int:
         return get_message_encoded_size(self._to_outgoing_amqp_message(self._message))
 
-    def encode_message(self):
+    def encode_message(self) -> bytes:
         output = bytearray()
         # to maintain the same behavior as uamqp, app prop values will not be decoded
-        self.application_properties = self._message.application_properties.copy()
+        if isinstance(self.application_properties, dict):
+            # casting Optional to Dict for copy
+            self.application_properties = cast(
+                Dict[Union[str, bytes], Any], self._message.application_properties).copy()
```

Comment:
I think the cast can be removed here by checking if `self._message.application_properties` is an instance of dict instead

Reply:
I tried that and it didn't work for some reason 🙁

Reply:
fixed, thanks Kashif!